src/dshould/package.d

```diff
@@ -0,0 +1,130 @@
+module dshould;
+
+import dshould.ShouldType;
+
+public import dshould.basic;
+public import dshould.contain;
+public import dshould.empty;
+public import dshould.stringcmp;
+public import dshould.thrown;
+
+// dispatch based on type
+void equal(Should, T)(Should should, T value, string file = __FILE__, size_t line = __LINE__)
+if (isInstanceOf!(ShouldType, Should))
+{
+    static if (is(typeof(should.got()) == string) && is(T == string) && !Should.hasWord!"not")
+    {
+        dshould.stringcmp.equal(should, value, file, line);
+    }
+    else
+    {
+        dshould.basic.equal(should, value, file, line);
+    }
+}
+
+auto equal(Should)(Should should)
+if (isInstanceOf!(ShouldType, Should))
+{
+    return should.basic.equal(should);
+}
+
+template throwA(T : Throwable)
+{
+    void throwA(Should, string file = __FILE__)(Should should, string msgTest, size_t line = __LINE__)
+    if (isInstanceOf!(ShouldType, Should))
+    {
+        dshould.thrown.throwA!T(should, file, line).where.its.msg.should.equal(msgTest, file, line);
+    }
+
+    auto throwA(Should, string file = __FILE__)(Should should, size_t line = __LINE__)
+    if (isInstanceOf!(ShouldType, Should))
+    {
+        return dshould.thrown.throwA!T(should, file, line);
+    }
+}
+
+alias throwAn = throwA;
+
+@("compares messages in throwA string overload")
```

Comment:
most of the following unit tests look like they are in the wrong place:
`package.d` should just contain `public import` and some dispatching
and only unit-test cases related to this dispatching

Reply:
The unittests here need more than one module. They're placed here to avoid circular dependencies.